tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -3626,52 +3626,187 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id, mdMet
+        sizeof(inner_signature), 0, 0, &inner_method_token);
+    if (FAILED(hr))
+    {
+        Logger::Warn("GenerateVoidILStartupMethod: DefineMethod failed for __DDInvokeLoader__");
         return hr;
     }
 
     // ****************************************************************************************************************
-    // Locals signature
+    // Standalone signature
     // ****************************************************************************************************************
 
     /////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // Generate the signature used to call the native logging method
+    COR_SIGNATURE log_signature[] = {
+        IMAGE_CEE_CS_CALLCONV_STDCALL, // Calling convention
+        3,                             // Number of parameters
+        ELEMENT_TYPE_VOID,             // Return type
+        ELEMENT_TYPE_I4,               // First parameter type
+        ELEMENT_TYPE_I,                // Second parameter type
+        ELEMENT_TYPE_I4,               // Second parameter type
+    };
+
+    mdSignature log_signature_token;
+    metadata_emit->GetTokenFromSig(log_signature, sizeof(log_signature), &log_signature_token);
+
+    // ****************************************************************************************************************
+    // Outer method
+    // ****************************************************************************************************************
+
+    {
+        /////////////////////////////////////////////
+        // Generate a locals signature defined in the following way:
+        //   [0] System.String (exception message)
+        //   [1] char*         (pointer to the exception message)
+        //   [2] char[] pinned (pinned char array for the exception message)
+        mdSignature locals_signature_token;
+        COR_SIGNATURE locals_signature[] = {
+            IMAGE_CEE_CS_CALLCONV_LOCAL_SIG, // Calling convention
+            3,
+            ELEMENT_TYPE_STRING,
+            ELEMENT_TYPE_PTR,
+            ELEMENT_TYPE_CHAR,
+            ELEMENT_TYPE_PINNED,
+            ELEMENT_TYPE_SZARRAY,
+            ELEMENT_TYPE_CHAR,
+        };
+        hr = metadata_emit->GetTokenFromSig(locals_signature, sizeof(locals_signature), &locals_signature_token);
+        if (FAILED(hr))
+        {
+            Logger::Warn("GenerateVoidILStartupMethod: Unable to generate outer locals signature. ModuleID=", module_id);
+            return hr;
+        }
+
+        ILRewriter rewriter(this->info_, nullptr, module_id, *ret_method_token);
+        rewriter.InitializeTiny();
+
+        rewriter.SetTkLocalVarSig(locals_signature_token);
+
+        if (FAILED(hr))
+        {
+            Logger::Warn("RunILStartupHook: Call to ILRewriter.Import() failed for outer method ", module_id, " ", ret_method_token);
```

Comment:
```suggestion
            Logger::Warn("RunILStartupHook: Call to ILRewriter.SetTkLocalVarSig() failed for outer method ", module_id, " ", ret_method_token);
```

Reply:
Oh yeah, I removed the call to `Import` but forgot to remove the log 🙈